src/addressindex.h

```diff
@@ -11,6 +11,16 @@
 
 #include <chrono>
 
+namespace AddressType {
+enum AddressType {
+    P2PK = 1,
+    P2PKH = 1,
```

Comment:
Using two different types for the same ID was the behavior in existing code and the refactors, since they're aiming to be backwards compatible, didn't modify that.

> even if they have same processing logic

They don't*, and furthermore, because they're reusing the same value, I have to create `TrimScriptP2{PK,PKH,SH}` instead of just creating `TrimScript` that branches based on `AddressType`. 

I plan on giving them distinct values in a PR that would, regardless of proposed change, be introducing breaking changes (i.e. requiring reindexing) but that isn't this PR. 

*\* - P2PK transactions don't actually correspond to an address, addresses are just ways of representing a 160-bit blob with some data tacked on before and after for metadata and checksumming respectively and a public key is longer than that*

*I assume, for simplicity's sake (or for Insight), the P2PK transactions are being represented as being received to their P2PKH equivalent address and creating that takes an extra step*

Reply:
Hm, but if they are serialized in same value and used by 3rd party service (is it true?), you should not refactor it to change their values after serialization.

Our logic - our pain, but for compatibility we should preserve value "one" for both. So, let's better to remove duplicate somehow and keep logic with comprehensive name "pk_or_pkh"

Reply:
I don't believe their serialization would affect Insight, since Insight should only be concerned with RPC output. The changes I'm proposing are more in line towards, at some point in the future, reworking the indexes to use Bitcoin's indexing APIs, allowing it to better integrate into the codebase as opposed to sticking out and being something we have to always work around instead of working with.

But that's in the future. For now, changed to `P2PK_OR_P2PKH`, resolved in latest push.